chapters/01-preface.Rmd

```diff
@@ -0,0 +1,334 @@
+applications they are being asked to support have shifted from fluid dynamics to
+data analysis; this guide will teach them how to build and run re-usable data
+analysis pipelines so that they can pass those skills on to their users.
+
+### Prerequisites
+
+Learners are expected to be able to:
+
+- Write reusable functions to analyze tabular data using [R][r].
+- Have a basic understanding of [Git][git] and [GitHub][github], and have at least
+used them for basic version control tasks.
+- Create reproducible reports using [R Markdown][r-markdown].
+
+Learners will need:
+
+- A personal computer with Internet access.
+- To have [R][r], [Git][git], and [RStudio][rstudio] installed.
+- A [GitHub][github] account.
+
+### What's the big picture?
+
+This book and the tools, skills, and knowledge being taught come out of the
+growing awareness of and demand for [open science][open-science],
+[reproducible research][reproducibility], and 
+[software sustainability][sustainable-software]. So what are these?
+
+- [Open science][open-science] is a movement to make scientific data, methods,
+and results publicly and freely accessible by publishing them with 
+[open licenses][open-license]. These licenses allow others to re-use and modify
+these outputs without worrying about copyright infringement, often only requiring
+attribution to the original source.
+- [Reproducible research][reproducible-research] means ensuring that anyone with
+access to data and software can feasibly reproduce the same or similar results,
+both to check them and to build on them. It's important to note that science can
+be open but not reproducible and reproducible but not open.
+- [Sustainable software][sustainable-software] is when software is developed in
+a way that makes it easier for other people to extend and/or maintain it, rather
+than to replace it. Sustainability isn't just about the software itself, it also
+depends on the skills and culture of its users. For instance, if a software
+package is being maintained by a couple of post-docs who are being paid a
+fraction of what they could earn in industry and have no realistic hope of
+promotion because their field doesn't value tool building, it doesn't matter
+whether the package is well tested and easy to install: sooner or later
+(probably sooner) it will become [abandonware][abandonware].
+
+### Overview of the packages created throughout course
+
+Throughout the book, you will build several packages. 
+
+- The main package used throughout the book is the Zipf's Law package `zipfs`.
+This package is used for the code-alongs, the actual content, and most of the
+exercises. More details about the Zipf's Law will be introduced in Chapter
+\@ref(package-setup).
+- The example package for the first project assignment is the Weather in Kenya
+package `kenyaweather`. This package is used as a reference when you work on
+your package for the first project assignment.
```

Comment:
We [defined](https://github.com/merely-useful/r-rse/issues/1#issue-639616529) the Kenya weather data package as the final/assignment package, but I'm not sure how to square this terminology against the following two bullet points.

Reply:
After reading the rest of this chapter as written, I'm wondering if we should even mention the other two assignment packages. We could include them in the instructor notes, and leave the chapter's language as unambiguous as possible.

Reply:
Great, I'll move them into the instructor section.